kqueen/engines/jenkins.py

```diff
@@ -321,3 +349,24 @@ def get_progress(self):
         except Exception:
             response = 500
         return {'response': response, 'progress': progress, 'result': result}
+
+    @classmethod
+    def engine_status(cls, **kwargs):
+        """
+        Implementation of :func:`~kqueen.engines.base.BaseEngine.engine_status`
+        """
+        conn_kw = {
+            'username': kwargs.get('username', config.get('JENKINS_USERNAME')),
+            'password': kwargs.get('password', config.get('JENKINS_PASSWORD')),
+            'timeout': 10
+        }
+        status = config.get('PROVISIONER_UNKNOWN_STATE')
+        try:
+            client = jenkins.Jenkins(config.get('JENKINS_API_URL'), **conn_kw)
+            auth_verify = client.get_whoami()
+            if auth_verify:
+                status = config.get('PROVISIONER_OK_STATE')
+        except Exception as e:
+            logger.exception('Could not contact JenkinsEngine backend: ')
```

Comment:
Looks like smth is missing

Reply:
It's just code movement, not a scope of this commit